game/MiiStudio_Miidatafile.ksy

```diff
@@ -0,0 +1,142 @@
+meta:
+  id: miidata_ms
+  endian: le
+seq:
+  - id: facial_hair_color
+    type: u1
+    doc: Facial hair color. Ranges from 0 to 99. Not ordered the same as visible in editor.
+  - id: beard_goatee
+    type: u1
+    doc: Beard (goatee) type. Ranges from 0 to 5.
+  - id: body_weight
+    type: u1
+    doc: Body weight. Ranges from 0 to 127, small to large.
+  - id: eye_stretch
+    type: u1
+    doc: Eye stretch. Ranges from 0 to 6, small to large.
+  - id: eye_color
+    type: u1
+    doc: Eye color. Ranges from 0 to 99. Not ordered the same as visible in editor.
+  - id: eye_rotation
+    type: u1
+    doc: Eye rotation. Ranges from 0 to 7, down to up. Note that some eye types have a default rotation.
+  - id: eye_size
+    type: u1
+    doc: Eye size. Ranges from 0 to 7, small to large.
+  - id: eye_type
+    type: u1
+    doc: Eye type. Ranges from 0 to 59. Not ordered the same as visible in editor.
+  - id: eye_horizontal
+    type: u1
+    doc: Eye X (horizontal) distance. Ranges from 0 to 12, close to far.
+  - id: eye_vertical
+    type: u1
+    doc: Eye Y (vertical) position. Ranges from 18 to 0, low to high.
+  - id: eyebrow_stretch
+    type: u1
+    doc: Eyebrow stretch. Ranges from 0 to 6, small to large.
+  - id: eyebrow_color
+    type: u1
+    doc: Eyebrow color. Ranges from 0 to 99. Not ordered the same as visible in editor.
+  - id: eyebrow_rotation
+    type: u1
+    doc: Eyebrow rotation. Ranges from 0 to 11, down to up. Note that some eyebrow types have a default rotation.
+  - id: eyebrow_size
+    type: u1
+    doc: Eyebrow size. Ranges from 0 to 8, small to large.
+  - id: eyebrow_type
+    type: u1
+    doc: Eyebrow type. Ranges from 0 to 23. Not ordered the same as visible in editor.
+  - id: eyebrow_horizontal
+    type: u1
+    doc: Eyebrow X (horizontal) distance. Ranges from 0 to 12, close to far.
+  - id: eyebrow_vertical
+    type: u1
+    doc: Eyebrow Y (vertical) distance. Ranges from 18 to 3, low to high.
+  - id: face_color
+    type: u1
+    doc: Skin color. Ranges from 0 to 9. Not ordered the same as visible in editor.
+  - id: face_makeup
+    type: u1
+    doc: Face makeup. Ranges from 0 to 11.
+  - id: face_type
+    type: u1
+    doc: Face shape. Ranges from 0 to 11. Not ordered the same as visible in editor.
+  - id: face_wrinkles
+    type: u1
+    doc: Face wrinkles. Ranges from 0 to 11.
+  - id: favorite_color
+    type: u1
+    doc: Favorite color. Ranges from 0 to 11.
+  - id: gender
+    type: u1
+    doc: Mii gender. 0 = male, 1 = female.

```

Comment:
It probably makes sense to define an [enum](https://doc.kaitai.io/user_guide.html#enums) for this field.

Actually, there are a few other fields in the format that could also use enums (colors, beard/eye/eyebrow/etc. types), but those fields have so many possible values and are probably difficult to describe in text sometimes, so it might not be worth the effort for those fields.

(Same for the `gender` fields in the other files.)